src/aleph_client/commands/instance/__init__.py

```diff
@@ -824,38 +824,40 @@ async def confidential_init_session(
 
     sevctl_path = find_sevctl_or_exit()
 
-    if (session_dir / "vm_godh.b64").exists():
-        if keep_session is None:
-            keep_session = Confirm.ask(
-                "Session already initiated for this instance, are you sure you want to override the previous one? You won't be able to communicate with already running VM"
-            )
+    client = VmConfidentialClient(account, sevctl_path, domain)
+    godh_path = session_dir / "vm_godh.b64"
+
+    if godh_path.exists() and keep_session is None:
+        keep_session = not Confirm.ask(
+            "Session already initiated for this instance, are you sure you want to override the previous one? You won't be able to communicate with already running VM"
+        )
         if keep_session:
             echo("Keeping already initiated session")
-            return
 
-    client = VmConfidentialClient(account, sevctl_path, domain)
+    # Generate sessions certificate files
+    if not ((session_dir / "vm_godh.b64").exists() and keep_session):
 
-    code, platform_file = await client.get_certificates()
-    if code != 200:
-        echo("Could not get the certificate from the CRN.")
-        return 1
+        code, platform_file = await client.get_certificates()
+        if code != 200:
+            echo("Could not get the certificate from the CRN.")
```

Comment:
Could we have a error message with more information? This really just says "error, handle this shit yourself" and that's not a great UX imo :/ (but idk if we have more information that just the error code)